cpp/src/slice2java/JavaUtil.h

```diff
@@ -83,78 +92,67 @@ namespace Slice
         //
         // Convert a Slice scoped name into a Java name.
         //
-        [[nodiscard]] std::string convertScopedName(
+        [[nodiscard]] static std::string convertScopedName(
             const std::string&,
             const std::string& = std::string(),
-            const std::string& = std::string()) const;
+            const std::string& = std::string());
 
         //
         // Returns the package prefix of a Contained entity.
         //
-        [[nodiscard]] std::string getPackagePrefix(const ContainedPtr&) const;
+        [[nodiscard]] static std::string getPackagePrefix(const ContainedPtr&);
 
         //
         // Returns the Java package of a Contained entity.
         //
-        [[nodiscard]] std::string getPackage(const ContainedPtr&) const;
+        [[nodiscard]] static std::string getPackage(const ContainedPtr&);
 
         //
         // Returns the Java type without a package if the package
         // matches the current package
         //
-        [[nodiscard]] std::string getUnqualified(const std::string&, const std::string&) const;
+        [[nodiscard]] static std::string getUnqualified(const std::string&, const std::string&);
 
         //
         // Returns the Java name for a Contained entity. If the optional
         // package argument matches the entity's package name, then the
         // package is removed from the result.
         //
-        [[nodiscard]] std::string getUnqualified(
+        [[nodiscard]] static std::string getUnqualified(
             const ContainedPtr&,
             const std::string& = std::string(),
             const std::string& = std::string(),
-            const std::string& = std::string()) const;
+            const std::string& = std::string());
 
         //
         // Return the method call necessary to obtain the static type ID for an object type.
         //
-        [[nodiscard]] std::string getStaticId(const TypePtr&, const std::string&) const;
+        [[nodiscard]] static std::string getStaticId(const TypePtr&, const std::string&);
 
         //
         // Returns the optional type corresponding to the given Slice type.
         //
-        std::string getOptionalFormat(const TypePtr&);
```

Comment:
I added a `[[nodiscard]]` here. This function has no side-effects and returns a string.
So, it should probably have `[[nodiscard]]` right?